CorsixTH/Lua/dialogs/place_staff.lua

```diff
@@ -77,15 +77,31 @@ end
 
 local flag_cache = {}
 local flag_altpal = 16
+-- Is this coordinate a valid tile for this staff member?
+local function isValidPlace(self)
```

Comment:
If you only read `function isValidPlace(self)`, this function can mean just about anything. `self` is quite literally any object that exists, and "valid place" applies to almost all objects too in one way or another.
In other words, the names are too generic, it can be interpreted too wide. The function name and the parameter don't give enough clues what use is intended.

Your comment says it's for a staff member, the `self` is thus easily replaced by a better name, eg `staff_member`,
`isValidPlace(staff_member)` is already much better.

Valid places for staff members are complicated, it could check the staff member for being in the right room, or inside the hospital, or not in the toilet, or a bunch of other notions of 'valid' for staff.
I don't quite know what the function is supposed to do, but since this is `place_staff.lua`, I am guessing something like `canBeDroppedHere(staff_member)` ?

(see how much difference a few well-picked names make?)

One thing that still bothers me is that 'dropping' or 'here' implies needing a position, and the function has no position parameter. I tried wording it in the function name, but I couldn't come up with a good solution for it (maybe you can).
Given my failure, I would describe that in the lua doc comments now.

Reply:
`-- Is this staff member permitted to be placed where the user is hovering or clicking?`

It describes the bigger picture rather than just what that function is there to judge? The user's actions set the coordinates.

Reply:
Yeah, I messed up big time in understanding what the function was doing. Sorry about that.

As for your question, if you write an explanation in comment, there is only one spot where you can read it. If you document it in the function name, it's everywhere for free. This is why a good function name has a lot of value, and a good parameter name helps near the definition, and in the code (and in the generated documentation if you ever read that).

You changed it to
``` lua
-- Is this staff member permitted to be placed where the user is hovering or clicking?
local function isValidStaffPlacement(staff_member)
```
Given my confusion I am surprised you changed the parameter name to my (what I now believe, wrong) suggestion. Please don't assume I am always right. In fact, if you aren't convinced by my arguments, don't change things. I may have played the game of code changing more than you, but that doesn't make me a gospel of truth :) 
It's your code change, you must aim for what you believe is the best possible solution.

As the parameter was the window and not the staff member, I now believe `self` was the right name after all.

The general idea of the comment is fine. The comment part `this staff member` feels a bit confusing to me. It suggests to me there is more than one in the class (there is this staff and there is thus also other staff). Phrasing it as "the staff member" doesn't trigger the 'this/other' link for me.

I have another question, why is this function not a method of the UI class, ie
``` lua
function UIPlaceStaff:_isValidStaffPlacement()
```
This makes it fully clear that the function is part of the ui class. The leading `_` means it's a method intended for internal used (a private method). It's not in the coding conventions it seems, but we do have code that uses it.

Reply:
With the function and parameter names both options work for me.

For this/that I mildly agree. It's the one given to the function.